include/swift/AST/ASTPrinter.h

```diff
@@ -263,6 +263,14 @@ class ASTPrinter {
     printStructurePre(Kind, D);
   }
 
+  /// Make a callback to printStructurePost(), performing any necessary
+  /// bookkeeping.
+  void callPrintStructurePost(PrintStructureKind Kind,
+                              const Decl *D = nullptr) {
+    forceNewlines();
```

Comment:
Ah, ok. I was wondering why we didn't have this - I didn't realize we were sticking to a pattern.